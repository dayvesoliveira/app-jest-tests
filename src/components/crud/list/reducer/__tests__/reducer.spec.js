import { 
    FETCH_SEARCH_DATA,
    FETCH_SEARCH_SUCCESS, 
    FETCH_SEARCH_FAILURE,
    RESET_SEARCH_DATA,
    DELETE_REGISTER_DATA,
    DELETE_REGISTER_FAILURE
} from "../../action-creators"

import { search, remove, initialState } from '../index'
import { MOCK_POSTS } from '../../../../../config/tests/__mocks__/posts.mock'

const listPostsMock = MOCK_POSTS
const POST_MOCK = listPostsMock[0]

describe('#Posts reducers',()=>{

    describe('search', ()=>{

        it('should return the initial state by default', ()=>{
            expect(search([],{})).toMatchSnapshot()
        })

        it('should return the `FETCH_SEARCH_DATA` state', ()=>{
            expect(search({},{type: FETCH_SEARCH_DATA })).toMatchSnapshot()
        })

        it('should return the `FETCH_SEARCH_SUCCESS` state', ()=>{

            const actionMock = {
                type: FETCH_SEARCH_SUCCESS,
                list: MOCK_POSTS
            }

            expect(search([],actionMock)).toEqual({
                list: MOCK_POSTS,
                isLoading: false
            })
        })

        it('should return the `FETCH_SEARCH_FAILURE` state', ()=>{            
            expect(search([],{type: FETCH_SEARCH_FAILURE})).toMatchSnapshot()
        })

        it('should return the `RESET_SEARCH_DATA` state', ()=>{
           expect(search(initialState,{type: RESET_SEARCH_DATA})).toMatchSnapshot()
        })
    })

    describe('delete', ()=>{
        
        it('should return the `FETCH_SEARCH_DATA` state', ()=>{
            expect(remove(undefined,{type: FETCH_SEARCH_DATA })).toMatchSnapshot()
        })

        it('should return the `FETCH_SEARCH_SUCCESS` state', ()=>{
            expect(remove(undefined,{type: FETCH_SEARCH_SUCCESS })).toMatchSnapshot()
        })

        it('should return the `DELETE_REGISTER_DATA` state', ()=>{
            expect(remove({},{type: DELETE_REGISTER_DATA })).toMatchSnapshot()
        })

        it('should return the `DELETE_REGISTER_FAILURE` state', ()=>{            
            expect(remove([],{type: DELETE_REGISTER_FAILURE})).toMatchSnapshot()
        })
    })

})