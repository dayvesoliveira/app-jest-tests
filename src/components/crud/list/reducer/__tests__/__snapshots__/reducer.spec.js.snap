// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#Posts reducers delete should return the \`DELETE_REGISTER_DATA\` state 1`] = `Object {}`;

exports[`#Posts reducers delete should return the \`DELETE_REGISTER_FAILURE\` state 1`] = `
Object {
  "isLoading": false,
}
`;

exports[`#Posts reducers delete should return the \`FETCH_SEARCH_DATA\` state 1`] = `
Object {
  "error": Object {},
  "isLoading": true,
  "list": Array [],
}
`;

exports[`#Posts reducers delete should return the \`FETCH_SEARCH_SUCCESS\` state 1`] = `
Object {
  "error": Object {},
  "isLoading": false,
  "list": Array [],
}
`;

exports[`#Posts reducers search should return the \`FETCH_SEARCH_DATA\` state 1`] = `
Object {
  "isLoading": true,
}
`;

exports[`#Posts reducers search should return the \`FETCH_SEARCH_FAILURE\` state 1`] = `
Object {
  "isLoading": false,
}
`;

exports[`#Posts reducers search should return the \`RESET_SEARCH_DATA\` state 1`] = `
Object {
  "error": Object {},
  "isLoading": false,
  "list": Array [],
}
`;

exports[`#Posts reducers search should return the initial state by default 1`] = `Array []`;
